src/audio/source_api_helper.c

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2023 Intel Corporation. All rights reserved.
+//
+
+#include <sof/audio/source_api.h>
+#include <sof/audio/source_api_implementation.h>
+#include <sof/audio/audio_stream.h>
+
+void source_init(struct source __sparse_cache *source, const struct source_ops *ops,
+		 struct sof_audio_stream_params *audio_stream_params)
+{
+	source->ops = ops;
+	source->locked_read_frag_size = 0;
```

Comment:
usually when a function or a member uses the term 'locked' there's a lock involved?

This is confusion to me, what part is locked here? This is worthy of a comment, you can't just mention a lock and not explain what the design is.

Reply:
there's no locking at this level. Name indeed misleading, changed